discovery-infra/install_cluster.py

```diff
@@ -207,18 +208,47 @@ def run_installation_flow_kube_api(
         nodes_number,
         kubeconfig_path,
 ):
-    log.info("Approving agents")
-    for agent in cluster_deployment.wait_for_agents(nodes_number):
-        agent.approve()
+    try:
+        log.info("Approving agents")
+        for agent in cluster_deployment.wait_for_agents(nodes_number):
+            agent.approve()
 
-    log.info("Waiting for installation to start")
-    agent_cluster_install.wait_to_be_installing()
+        log.info("Waiting for installation to start")
+        agent_cluster_install.wait_to_be_installing()
 
-    log.info("Waiting until cluster finishes installation")
-    agent_cluster_install.wait_to_be_installed()
+        log.info("Waiting until cluster finishes installation")
+        agent_cluster_install.wait_to_be_installed()
 
-    log.info("Download kubeconfig-noingress")
-    agent_cluster_install.download_kubeconfig(kubeconfig_path=kubeconfig_path)
+        log.info("Download kubeconfig-noingress")
+        agent_cluster_install.download_kubeconfig(kubeconfig_path=kubeconfig_path)
+    except Exception as err:
+        log.error(f"Failure during kube-api installation flow ({err})")
+    finally:
+        collect_debug_info_from_cluster(cluster_deployment, agent_cluster_install)
+
+
+def collect_debug_info_from_cluster(cluster_deployment, agent_cluster_install):
+        cluster_name = cluster_deployment.ref.name
+        output_folder = f'build/{cluster_name}'
+        utils.recreate_folder(output_folder)
+
+        aci = agent_cluster_install.get()
+        debugInfo = aci['status']['debugInfo']
+        eventsURL = debugInfo['eventsURL']
```

Comment:
ideally, these field are not always returned in the debugInfo (only when have values)
maybe we should handle cases these keys do not exist.

Reply:
right, added a check for these.